scheduler/src/cook/mesos/api.clj

```diff
@@ -419,18 +449,43 @@
         :allowed? (fn [ctx]
                     (condp contains? (get-in ctx [:request :request-method])
                       #{:get :delete}
-                      (letfn [(used? [uuid]
-                                (try
-                                  (unused-uuid? (db conn) uuid)
-                                  false
-                                  (catch Exception e
-                                    true)))]
-                        (or (every? used? (::jobs ctx))
-                          [false {::error (str "UUID "
-                                               (str/join
-                                                 \space
-                                                 (remove used? (::jobs ctx)))
-                                               " didn't correspond to a job")}]))
+                      (let [user  (get-in ctx [:request :authorization/user])
+                            uuids (::jobs ctx)
+
+                            datomic (db conn)
+                            used? (fn [uuid]
+                                    (try
+                                      (unused-uuid? datomic uuid)
+                                      false
+                                      (catch Exception e
+                                        true)))
+
+                            ;; Changes each UUID into a map describing
+                            ;; whether it's allowed, and if not, why
+                            ;; it's not allowed.
+                            uuid-filter (fn [uuid]
+                                          (let [job (fetch-job-map (db conn)
+                                                                   fid
+                                                                   uuid)]
+                                            (if (or (nil? job) 
+                                                    (not (is-authorized? user :access job)))
+                                              {:is-allowed false
+                                               :message (str "No job found with UUID " uuid)
+                                               :uuid uuid}
+                                              {:is-allowed true
+                                               :uuid uuid})))
+
+                            uuids (map uuid-filter uuids)]
+                        ;; Return true if every UUID is in use, or
+                        ;; else return false with a list of
+                        ;; nonexistant UUIDs.
+                        (if (every? (map :is-allowed uuids))
+                          true
+                          (let [message  (->> (map :message uuids)
+                                                 (remove nil?)
+                                                 (str/join "; " ))]
+                            (log/info "Denying access:" message)
```

Comment:
If we are logging denying access, I would like to separate out the case when there simply was no job for that uuid.